### PR TITLE
feat(search): offline users cannot search archive

### PR DESCRIPTION
**Pull request by cyndichin**
## Summary
* Handles offline user scenario when the user navigates to the search experience before submitting search
* Also refactors how we were managing state for our `SearchView`

## References 
IN-962

## Implementation Details
Refactor how we were presenting the different states of the `SearchView`. Instead of managing three separate objects, I decided to create a single enum object called `SearchViewState` that can set which view should be shown and the associated values that will be used in presenting the details of the view. Added some logic of checking if the user is offline + show the offline view when they enter the search experience in `freeEmptyState` and `premiumEmptyState ` 

## Test Steps
**For premium user:**
- [ ] WHEN a user selects the “Saves” search scope
AND the device is offline
THEN show the recent searches view

**For premium and free user:**
 - [ ] WHEN a user selects the “Archive” search scope
AND the device is offline
THEN show a message in the search results area

”Your device is currently offline.” 
”You must have a network connection to search your archived items”

- [ ] WHEN a user submits a search
AND “Archive” is the selected scope
AND the device is offline
THEN show a message in the search results area

”Your device is currently offline.” 
”You must have a network connection to search your archived items”

**For premium user, free user should see Premium upsell instead**
- [ ] WHEN a user selects the “All items” search scope
AND the device is offline
THEN show a message in the search results area
”Your device is currently offline.” 
”You must have a network connection to search all items”

- [ ] WHEN a user submits a search
AND “All items” is the selected scope
AND the device is offline
THEN show a message in the search results area
”Your device is currently offline.” 
”You must have a network connection to search all items”

## PR Checklist:
- [x] Added Unit / UI tests
- [x] Self Review (review, clean up, documentation, run tests)
- [x] Basic Self QA

### CODE REVIEW

File: Pocket.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved [hunk 1]
```diff
@@ -140,8 +140,8 @@
       "kind" : "remoteSourceControl",
       "location" : "https://github.com/apple/swift-nio.git",
       "state" : {
-        "revision" : "4ad2c3733845abd9ee8892a323b0fa0d80f37e34",
-        "version" : "2.47.0"
+        "revision" : "45167b8006448c79dda4b7bd604e07a034c15c49",
+        "version" : "2.48.0"
```
Comment (timc-mozilla) [inline Pocket.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved:144]: have you merged in develop that fixes this?

Comment (cyndichin) [reply Pocket.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved:144]: this is an updated version compared to develop, not sure why it's updating now

Comment (cyndichin) [reply Pocket.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved:144]: https://github.com/Pocket/pocket-ios/blob/develop/Pocket.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved

Comment (cyndichin) [reply Pocket.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved:144]: could be that an exact version is relying on this dependency in the latest version as well

**Review by timc-mozilla — APPROVED** [review]

**Pull request merged**